src/test/atlas_planned_maintenance_testing/mod.rs

```diff
@@ -0,0 +1,114 @@
+mod json_models;
+
+use std::{
+    env,
+    fs::File,
+    io::{BufWriter, Write},
+    panic::AssertUnwindSafe,
+    path::PathBuf,
+};
+
+use futures::FutureExt;
+use serde_json::{from_str, from_value, Value};
+use time::OffsetDateTime;
+
+use crate::{
+    bson::{doc, Bson},
+    test::spec::unified_runner::{entity::Entity, test_file::TestFile, test_runner::TestRunner},
+};
+
+use json_models::{Events, Results};
+
+#[test]
+fn get_exe_name() {
```

Comment:
this is a slightly more convenient way to get the name of the test executable (as opposed to parsing through `cargo` output). all of the tests in the crate are run via one executable that can be filtered by passing test names as arguments. `std::env::current_exe()` returns the name of the executable that's currently running, so when it's called within a test it'll return the name of that test executable. this test method then writes that name to a file that the workload executor can invoke via `$(cat exe_name.txt workload_executor)`. (I used `into_os_string` on the `PathBuf` returned rather than the `Debug` implementation because the latter includes quotations around the path that would need to be stripped out.)